src/components/EventModal/Collect/index.vue

```diff
@@ -747,8 +776,9 @@ export default {
             );
             const result = await this.collectNFTWithLIKE(classId, {
               memo: this.memo,
+              email: this.email,
             });
-            if (result) {
+            if (result && !this.nftIsNFTBook) {
```

Comment:
Just check for `result.nftId || result.purchased?.[0]?.nftId` is undefined instead of hardcoding book?